#166265842 sign-up validation

PelumiAlesh opened the pull request:
### What does this PR do?
Add validation to the sign-up endpoint
#### Description of Task to be completed?
When a user tries to sign up with the wrong or unsupported input format, the user should receive descriptive signup/Registration validation error(s)
#### How should this be manually tested?
>- Clone the repo
>- Run `npm run dev`
>-  Make a `POST` request with bad input format to localhost:6000/api/v1/auth/signup
#### Any background context you want to provide?
No
#### What are the relevant pivotal tracker stories?
[#166265842](https://www.pivotaltracker.com/story/show/166265842)
#### Screenshots (if appropriate)
![Screenshot 2019-05-31 at 11 41 25 AM](https://user-images.githubusercontent.com/29557902/58700962-840a1900-8399-11e9-8b7c-b9dc308cb1c9.png)
#### Questions:

cvjude committed:
feat(Signin): Implement Social media Signin

>- Use passport access_token verification
>- Create facebook app, google app, twitter app

[Delivers #166265916]

PelumiAlesh committed:
feat(validation): write sign up validation

Write validation that returns descriptive message when user sign up with incorrect format of input

[Finishes #166265842]

PelumiAlesh committed:
feat(validation): write sign up validation

Write validation that returns descriptive message when user sign up with incorrect format of input

[Finishes #166265842]

PelumiAlesh committed:
feat(validation): write sign up validation

Write validation that returns descriptive message when user sign up with incorrect format of input

[Finishes #166265842]

PelumiAlesh committed:
bg(validations): Fix coverage failing bug

PelumiAlesh committed:
bg(validations): Fix coverage failing bug

PelumiAlesh committed:
bg(validations): Fix coverage failing bug

PelumiAlesh committed:
feat(validation): write sign up validation

Write validation that returns descriptive message when user sign up with incorrect format of input

[Finishes #166265842]

PelumiAlesh committed:
feat(validation): write sign up validation

Write validation that returns descriptive message when user sign up with incorrect format of input

[Finishes #166265842]

PelumiAlesh committed:
feat(validation): write sign up validation

Write validation that returns descriptive message when user sign up with incorrect format of input

[Finishes #166265842]

PelumiAlesh committed:
Fix test bug

PelumiAlesh committed:
Fix test bug

PelumiAlesh committed:
Fix test bug

chuxmykel commented on server/__test__/test.spec.js:
```diff
@@ -53,4 +53,28 @@ describe('test for user signup', () => {
+  beforeEach(async () => {
+       await truncate();
+      });
+  it('Should reigister a user when all required input is supplied', async (done) => {
```
I'm sure this isn't supposed to be the title of this `it` block.

chuxmykel reviewed (dismissed):
Check the title of that it block and you're good to go

PelumiAlesh committed:
bg(test): Fix test title

PelumiAlesh committed:
bg(test): Fix test title

PelumiAlesh committed:
bg(test): Fix test title

encodedBicoding committed:
feedback implementation

encodedBicoding committed:
Merge pull request #19 from emp-daisy/ft-integrate-social-media-login-166265916

#166265916 Implement Social media Sign in

ocranbillions reviewed (dismissed):
Handle the merge conflicts and its good to go.

chuxmykel reviewed (dismissed):
Work on merge conflict and you're good to go.

encodedBicoding committed:
feat(email-verify):
setup email verification
[(Finishes) #166265849]

encodedBicoding committed:
Remove console.log statements

encodedBicoding committed:
Update heroku link

cvjude committed:
Merge pull request #15 from emp-daisy/feat/166265849/verify-email-address

#166265849 Set up email address verification

chuxmykel committed:
feat(password reset): implement password reset

- implement password reset logic and integrate with mailing module

[Delivers #166265865]

encodedBicoding committed:
Merge pull request #22 from emp-daisy/ft/166265865/user-password-reset

#166265865 Implement password reset

encodedBicoding reviewed (commented):
Gone through the work, it is good. Just needs to fix merge conflicts.

ocranbillions committed:
feature(signin): Add email signin

    Add /signin enpoint to routes
    Add validation middleware for user inputs
    Add signinController as route handler
    Get email, password from request body
    Query db for user
    Return response based on the query result
[Delievers #166266544]

encodedBicoding committed:
Merge pull request #23 from emp-daisy/ft/166266544/email-sign-in

#166266544 Add email signin

PelumiAlesh committed:
bug(validation): rebase and fix conflicts

PelumiAlesh committed:
bug(validation): rebase and fix conflicts

encodedBicoding reviewed (changes requested):
Why is travis failing?

encodedBicoding reviewed (commented):
that sequelize table issh is causing build to fail. Are you using postgresDB or elephant?.

PelumiAlesh commented:
> that sequelize table issh is causing build to fail. Are you using postgresDB or elephant?.

Im using postgresDB

emp-daisy commented:
Why is this still pending?

PelumiAlesh commented:
THIS IS THE OLD BRANCH
